aleph_alpha_client/completion.py

```diff
@@ -0,0 +1,116 @@
+from typing import Any, Dict, List, Mapping, NamedTuple, Optional, Sequence, Union
+
+from aleph_alpha_client.image import ImagePrompt
+from aleph_alpha_client.prompt import _to_serializable_prompt
+
+
+class CompletionRequest(NamedTuple):
+    """
+    Describes a completion request
+
+    Parameters:
+        prompt (str, optional, default ""):
+            The text to be completed. Unconditional completion can be started with an empty string (default). The prompt may contain a zero shot or few shot task.
+
+        maximum_tokens (int, optional, default 64):
+            The maximum number of tokens to be generated. Completion will terminate after the maximum number of tokens is reached. Increase this value to generate longer texts. A text is split into tokens. Usually there are more tokens than words. The summed number of tokens of prompt and maximum_tokens depends on the model (for luminous-base, it may not exceed 2048 tokens).
```

Comment:
The _maximal supported_ (?) summed number of tokens of prompt and maximum_tokens depends ...

Reply:
This is not the maximal supported number. The maximal supported number of tokens is fixed and defined by the model.
Here, the user determines how many tokens he wants to have generated at most.
So it's more like the _maximal desired_ number of tokens.